packages/lesmis/src/components/EntityMenu.js

```diff
@@ -21,13 +31,131 @@ const TextButton = styled(MuiButton)`
 `;
 
 const Body = styled.div`
-  background: ${props => props.theme.palette.grey['100']};
   height: 100%;
-  padding: 1.8rem;
 `;
 
-export const EntityMenu = ({ buttonText }) => {
+const ContainerList = styled(MuiList)`
+  padding: 0;
+  background: ${COLORS.GREY_F9};
+
+  > li {
+    padding-left: 1.875rem;
+  }
+
+  > li:first-child {
+    border-top: none;
+  }
+`;
+
+const List = styled(MuiList)`
+  background: white;
+  padding: 0 0 0 6rem;
+`;
+
+const IconButton = styled(MuiIconButton)`
+  color: ${props => props.theme.palette.text.tertiary};
+
+  &:hover {
+    color: ${props => props.theme.palette.text.secondary};
+  }
+
+  svg.MuiSvgIcon-root {
+    width: 1.875rem;
+    height: 1.875rem;
+  }
+`;
+
+const ListItem = styled(MuiListItem)`
+  display: flex;
+  align-items: center;
+  padding: 0 1.875rem 0 0;
+  border-top: 1px solid ${props => props.theme.palette.grey['400']};
+
+  svg.place-icon {
+    width: 3.75rem;
+    height: 3.75rem;
+  }
+`;
+
+const ListItemText = styled(Typography)`
+  display: inline-block;
+  padding-top: 2.3rem;
+  padding-bottom: 2.3rem;
+  text-decoration: none;
+  flex: 1;
+  color: initial;
+  font-size: 1.125rem;
+  line-height: 1.4;
+  margin-left: 1rem;
+  transition: color 0.1s ease;
+
+  &:hover {
+    color: ${props => props.theme.palette.primary.main};
+  }
+`;
+
+const ListItemLink = props => <ListItemText component={RouterLink} {...props} />;
+
+const getEntitiesByCodes = (entities, codes) => entities.filter(e => codes.includes(e.code));
+
+const ListItemComponent = React.memo(({ entities, entity, onMenuClose }) => {
+  const [open, setOpen] = useState(false);
+  const hasChildren = Array.isArray(entity.childCodes);
```

Comment:
Are we doing any translation of the entity object when pulling from the `entity-server`? Otherwise I think this might be: `entity.child_codes`?

Reply:
Yes we are doing translation to camel case in lesmis-server connections. If you look in EntityConnection for example, you should see the translation there where we are leveraging the `camelcase-keys` library.